airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2854,15 +2854,16 @@ def rowcount(self) -> int:
         return -1
 
     def execute(self, operation: str, parameters: dict | None = None) -> None:
-        """Execute a BigQuery query, and return the job ID.
+        """Execute a BigQuery query, and update the BigQueryCursor description.
 
         :param operation: The query to execute.
         :param parameters: Parameters to substitute into the query.
         """
         sql = _bind_parameters(operation, parameters) if parameters else operation
         self.flush_results()
-        self.job_id = self._run_query(sql)
-
+        job = self._run_query(sql)
+        self.job_id = job.job_id
+        self.location = job.location if not self.location else self.location
```

Comment:
```suggestion
        self.location = self.location or job.location
```

Reply:
What maks said or in any case `Improve Readability With Positive Booleans`